conan/conan.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 Sonatype Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package conan
+
+import (
+    "github.com/package-url/packageurl-go"
+    "io/ioutil"
+    "os"
+    "path/filepath"
+    "strings"
+)
+
+var (
+    newline = "\n"
+)
+
+type conanPurlInfo struct {
+    name string
+    version string
+}
+
+type Options struct {
+    Directory  string
+    BinaryName string
+}
+
+type ConanGenerator struct {
+    filepath string
+}
+
+func New(options Options) *ConanGenerator {
+    if options.Directory == "" {
+        options.Directory = "."
+    }
+
+    return &ConanGenerator{
+        filepath: filepath.Join(options.Directory, "conanfile." + options.BinaryName + ".cheque"),
+    }
+}
+
+func (c ConanGenerator) CheckOrCreateConanFile(purls []packageurl.PackageURL) {
+    _, err := os.Stat(c.filepath)
```

Comment:
So for this one, I think you want to check if err is not os.IsNotExistErr or something like that (because the error could be something else). 

```
if _, err := os.Stat("/path/to/whatever"); os.IsNotExist(err) {
  // error happened, can't Stat!
  // /path/to/whatever does not exist
}
```
More or less that. If the error is something else, I would suggest returning it so:

`func (c ConanGenerator) CheckOrCreateConanFile(purls []packageurl.PackageURL) (err error) {`